common/scylla_configure.py

```diff
@@ -85,10 +84,22 @@ def save_scylla_yaml(self):
     def instance_user_data(self):
         if self._instance_user_data is None:
             try:
-                raw_user_data = self.cloud_instance.user_data
+                raw_user_data = self.cloud_instance.user_data.strip()
                 LOGGER.info("Got user-data: %s", raw_user_data)
-                self._instance_user_data = json.loads(raw_user_data) if raw_user_data.strip() else {}
-                LOGGER.debug("JSON parsed user-data: %s", self._instance_user_data)
+
+                # Try reading mime multipart message, and extract
+                # scylla-machine-image configuration out of it
+                message = message_from_string(raw_user_data)
+                if message.is_multipart():
+                    for part in message.walk():
+                        if part.get_content_type() in ('x-scylla/json', 'x-scylla/yaml'):
+                            raw_user_data = part.get_payload()
```

Comment:
You have a bug here:
The `raw_user_data` gets overwritten when we have more than one part of the `json/yaml` types.

Reply:
As a user, what would you expect, if you have passed in multiple configuration files ?

let say I've read 3 files, what should I do with them ? whom is the correct one ? do I patch them all together ?

I would rather leave it as last one wins

seems like cloud-init face a similar thing:
https://cloudinit.readthedocs.io/en/latest/topics/merging.html

I don't think we should implement anything like that now (or ever)

Reply:
If it is intentional, then, please, add an inline comment here in the code describing the intention.